things/service/src/main/java/org/eclipse/ditto/things/service/persistence/actors/strategies/commands/MigrateThingDefinitionStrategy.java

```diff
@@ -0,0 +1,279 @@
+
+
+    public MigrateThingDefinitionStrategy(final ActorSystem actorSystem) {
+        super(MigrateThingDefinition.class, actorSystem);
+    }
+
+    @Override
+    protected Result<ThingEvent<?>> doApply(final Context<ThingId> context,
+            @Nullable final Thing thing,
+            final long nextRevision,
+            final MigrateThingDefinition command,
+            @Nullable final Metadata metadata) {
+
+        final Thing existingThing = getEntityOrThrow(thing);
+        final Instant eventTs = getEventTimestamp();
+
+        return handleUpdateDefinition(context, existingThing, eventTs, nextRevision, command, metadata)
+                .toCompletableFuture()
+                .join();
+    }
+
+    private CompletionStage<Result<ThingEvent<?>>> handleUpdateDefinition(final Context<ThingId> context,
+            final Thing existingThing,
+            final Instant eventTs,
+            final long nextRevision,
+            final MigrateThingDefinition command,
+            @Nullable final Metadata metadata) {
+
+        final DittoHeaders dittoHeaders = command.getDittoHeaders();
+
+        // 1. Evaluate Patch Conditions and modify the migrationPayload
+        final JsonObject adjustedMigrationPayload = evaluatePatchConditions(
+                existingThing,
+                command.getMigrationPayload(),
+                command.getPatchConditions(),
+                dittoHeaders);
+
+        // 2. Generate Skeleton using definition
+        final CompletionStage<Thing> skeletonStage = generateSkeleton(context, command, dittoHeaders);
+
+        // 3. Merge Skeleton with Existing Thing and Apply Migration Payload
+        final CompletionStage<Thing> updatedThingStage = skeletonStage.thenApply(skeleton -> {
+            final Thing mergedThing = mergeSkeletonWithThing(existingThing, skeleton, command.getThingDefinitionUrl(), command.isInitializeMissingPropertiesFromDefaults());
+            return applyMigrationPayload(mergedThing, adjustedMigrationPayload, dittoHeaders, nextRevision, eventTs);
+        });
+
+        // 4. Validate and Build Result
+        return updatedThingStage.thenCompose(updatedThing ->
+                buildValidatedStage(command, existingThing, updatedThing)
+                        .thenCompose(validatedCommand -> {
+                            final MergeThing mergeThingCommand = MergeThing.of(
+                                    command.getEntityId(),
+                                    JsonPointer.empty(),
+                                    updatedThing.toJson(),
+                                    dittoHeaders
+                            );
+
+                            final MergeThingStrategy mergeStrategy = new MergeThingStrategy(context.getActorSystem());
+                            final Result<ThingEvent<?>> mergeResult = mergeStrategy.doApply(
+                                    context,
+                                    existingThing,
```

Comment:
Why apply the mergeStrategy for the `existingThing` here?
I would expect the `updatedThing`.

So currently this e.g. validates the old, existingThing (as part of the MergeThingStrategy) against the WoT model instead of the new one.